lib/Backends/Interpreter/InterpreterDeviceManager.h

```diff
@@ -42,9 +42,14 @@ class InterpreterDeviceManager : public QueueBackedDeviceManager {
 public:
   InterpreterDeviceManager(std::unique_ptr<DeviceConfig> config = nullptr,
                            size_t maxMemory = 2000000000)
-      : QueueBackedDeviceManager(BackendKind::Interpreter, std::move(config)),
-        maxMemoryBytes_(maxMemory) {}
-
+      : QueueBackedDeviceManager(BackendKind::Interpreter, std::move(config)) {
+    if (maxMemory == 2000000000 && config_ != nullptr &&
```

Comment:
for brevity we could use, config_->getDeviceMemory(/*default*/maxMemory) and avoid all these checks across all device managers.

also, what are the scenarios when we create device manager with nullptr config?

Reply:
> for brevity we could use, config_->getDeviceMemory(/_default_/maxMemory) and avoid all these checks across all device managers.
> 
> also, what are the scenarios when we create device manager with nullptr config?

I saw sometimes the `createDeviceManger` is called with only a backendkind parameter. So I thought the device config might be set to nullptr which is its default value.  Should I keep this or just ignore the nullptr?.  Also could you clarify the ` config_->getDeviceMemory(/default/maxMemory)` :)?

Reply:
> for brevity we could use, config_->getDeviceMemory(/_default_/maxMemory) and avoid all these checks across all device managers.
> 
> also, what are the scenarios when we create device manager with nullptr config?

OK,  I merged all the backend if/else branches into the config_->getDeviceMemory() function. 
As we discussed this morning, I still check if config_ is a nullptr.